pkgs/top-level/all-packages.nix

```diff
@@ -1345,7 +1345,18 @@ with pkgs;
 
   substituteAllFiles = callPackage ../build-support/substitute-files/substitute-all-files.nix { };
 
-  replaceDependency = callPackage ../build-support/replace-dependency.nix { };
+  replaceDependencies = callPackage ../build-support/replace-dependencies.nix { };
+
+  replaceDependency = { drv, oldDependency, newDependency, verbose ? true }: replaceDependencies {
+    inherit drv verbose;
+    replacements = [{
+      inherit oldDependency newDependency;
+    }];
+    # When newDependency depends on drv, instead of causing infinite recursion, keep it as is.
+    cutoffPackages = [ newDependency ];
+  };
+
+  replaceDirectDependencies = callPackage ../build-support/replace-direct-dependencies.nix { };
```

Comment:
That's basically the way it was before, see also the sixth commit. Reasons are both that `replaceDirectDependencies` can be useful on its own for some quick hacks without needing IFD, and that it can easily be swapped out with a different implementation. I have made use of both in the past (currently only the latter).